wgpu-hal/src/gles/adapter.rs

```diff
@@ -8,7 +8,14 @@ use crate::auxil::db;
 
 const GL_UNMASKED_VENDOR_WEBGL: u32 = 0x9245;
 const GL_UNMASKED_RENDERER_WEBGL: u32 = 0x9246;
-
+extern "C" {
+    /// returns 1 if success. 0 if failure. extension name must be null terminated
+    pub fn emscripten_webgl_enable_extension(
+        context: std::ffi::c_int,
+        extension: *const std::ffi::c_char,
+    ) -> std::ffi::c_int;
+    pub fn emscripten_webgl_get_current_context() -> std::ffi::c_int;
+}
```

Comment:
Could we move this to a emscripten module, and expose from that module a simple enable_extension that uses them

Reply:
done